### PR TITLE
Create eip-1538

### PR DESCRIPTION
**Pull request by mudgen**
This pull request adds eip-1538. 

Information about ERC1538 is here: https://github.com/ethereum/EIPs/issues/1538

### CODE REVIEW

**Comment by fulldecent** [comment]
`_functionSignatures` may be better as a `bytes4[]`. Currently it is `string` which does not seem semantic.

**Comment by fulldecent** [comment]
Ownership is orthogonal to the goals of this project so I believe it is unnecessary for the specification to include this.

Also, there might be a vulnerability about storage.

I don't think ERC-930 is ready for prime time. More discussion there.

**Comment by mudgen** [comment]
> `_functionSignatures` may be better as a `bytes4[]`. Currently it is `string` which does not seem semantic.

Yes, using `bytes4[]` is semantically better but then functions in the ERC1538Query interface, as given in the standard, will not work. Because they require the function signatures.

Also, using strings of function signatures enables implementors to prevent a problem known as selector clashing,  where one signature hashes to the same 4 bytes as another.

**Comment by mudgen** [comment]
> Ownership is orthogonal to the goals of this project so I believe it is unnecessary for the specification to include this.

I agree with you which is why Ownership is not part of the standard. The standard says the following: "Typically some kind of authentication is needed for adding/replacing/removing functions from a transparent contract, however the scheme for authentication or ownership is not part of this standard."

Your comment:

> Also, there might be a vulnerability about storage.

If you know of a vulnerability then please give the details. 

>I don't think ERC-930 is ready for prime time. More discussion there.

Do you mean ERC1538?

**Comment by James-Sangalli** [comment]
Hi @mudgen would be good if in future the commit messages were more clear in their intent

**Comment by mudgen** [comment]
@James-Sangalli I understand, I will work on improving my commit messages.

**Comment by mudgen** [comment]
@fulldecent I appreciate your help and input about ERC1538. If there are ways to make it better then let's make it better. Because of your input and other's input the standard is getting better.  I recently added several new sections to the standard that you may be interested in. 

Here are titles of the new sections:
Security
Transparency
Versions of Functions
Best Practices, Tools and More Information
String of Function Signatures Instead of bytes4[] Array of Function Selectors

**Comment by fulldecent** [comment]
I do not see this new version in EIPs. Specifically I do not see "# String of Function"

**Comment by mudgen** [comment]
@fulldecent sorry, it wasn't updated in the EIP. I just updated it, so the new things are there now.

File: EIPS/eip-1538.md [hunk 1]
```diff
@@ -0,0 +1,453 @@
+---
+eip: 1538
+title: Transparent Contract Standard
+author: Nick Mudge <nick@mokens.io>
+status: Draft
+type: Standards Track
+category ERC
```
Comment (axic) [inline EIPS/eip-1538.md]: Misses a colon.

File: EIPS/eip-1538.md [hunk 2]
```diff
@@ -0,0 +1,453 @@
...
+created: 31 October 2018
```
Comment (axic) [inline EIPS/eip-1538.md]: This date format isn't following the required one, yyyy-mm-dd.

**Comment by axic** [comment]
@mudgen any chance this PR could be rebased? It seems like it imports a lot of unnecessary commits.

**Pull request closed without merging**